python/lsst/afw/detection/footprintSet.cc

```diff
@@ -102,6 +102,18 @@ PYBIND11_PLUGIN(_footprintSet) {
     clsFootprintSet.def("swap", &FootprintSet::swap);
     //    clsFootprintSet.def("swapFootprintList", &FootprintSet::swapFootprintList);
     clsFootprintSet.def("setFootprints", &FootprintSet::setFootprints);
+    // The pybind11 wrapped getFootprints dereferences the shared pointer to the std::vector of Footprints
+    // held by the FootprintSet when returning to python to work around a stl caster issue. However this
+    // creates a problem for the setter that now expects a shared pointer, making the setter unable to
+    // operate on the output form the getter. Below defines two overrides for the setter function, one
+    // that can handle a shared pointer (in case somewhere else in the stack has one somehow) and one
+    // that will accept a standard vector of Footprints, which will be internally placed in a shared
+    // pointer to match the underlying C++ function signature
+    clsFootprintSet.def("setFootprints",
+        [](FootprintSet &self, std::shared_ptr<FootprintSet::FootprintList> footListPtr)
+        {self.setFootprints(footListPtr);});
```

Comment:
To the former point, there is no difference to the standard &FootprintSet::setFootprints. Where the difference lies is that if you have two function signatures, you must do a function pointer cast to disambiguate. So I could have written (void (FootprintSet::*)(std::shared_ptr<FootprintSet::FootprintList>)) &FootprintSet::setFootprints but for the sake of symmetry with the other argument (and easier readability) I opted to have them both as lambda calls.

As for the circumstances for which someone could hold a pointer to a FootprintList? I am not sure, I don't actually think there is any (at least that I can think up) but I worried that there is somewhere and somebody will exercise the code at some point and we will have a new breakage. So I hedged against that and included it, as it doesn't really add much overhead to the bindings anyway.